Console/VaultMigrate.php

```diff
@@ -0,0 +1,393 @@
+    {
+        $select = $db->select()
+            ->join('customer_entity', 'customer_entity.entity_id = customer_entity_varchar.entity_id')
+            ->where(self::ATTRIBUTE_ID . ' = ?', $eavAttributeId)
+            ->from(self::CUSTOMER_ENTITY_TABLE, self::VALUE . ' as braintree_id');
+        $result = $db->fetchAll($select);
+
+        if ($result) {
+            $output->writeln('<info>'. count($result) .' stored cards found</info>');
+            return $result;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param OutputInterface $output
+     * @param $storedCards
+     * @return array
+     * @throws NotFound
+     */
+    private function findBraintreeCustomers(OutputInterface $output, $storedCards): array
+    {
+        $customers = [];
+        foreach ($storedCards as $storedCard) {
+            $output->writeln('<info>Search Braintree for Customer ID ' . $storedCard['braintree_id'] . '...</info>');
+            $customers[] = $this->braintreeAdapter->getCustomerById($storedCard['braintree_id']);
+        }
+
+        return $customers;
+    }
+
+    /**
+     * @param $customers
+     * @return array
+     */
+    public function remapCustomerData($customers): array
+    {
+        $remappedCustomerData = [];
+
+        foreach ($customers as $customer) {
+            $customerData = [
+                'braintree_id' => $customer->id,
+                'email' => $customer->email
+            ];
+
+            if ($customer->creditCards) {
+                // grab each stored credit card
+                foreach ($customer->creditCards as $creditCard) {
+                    $customerData['storedCards'][] = [
+                        'token' => $creditCard->token,
+                        'expirationMonth' => $creditCard->expirationMonth,
+                        'expirationYear' => $creditCard->expirationYear,
+                        'last4' => $creditCard->last4,
+                        'cardType' => self::CC_MAPPER[str_replace(' ', '-', strtolower($creditCard->cardType))]
+                    ];
+                }
+            }
+
+            // Add customer data to the main customer array
+            $remappedCustomerData[] = $customerData;
+        }
+
+        return $remappedCustomerData;
+    }
+
+    /**
+     * @param OutputInterface $output
+     * @param array $customers
+     */
+    private function migrateStoredCards(OutputInterface $output, array $customers)
+    {
+        $websites = $this->storeManager->getWebsites();
+
+        foreach ($websites as $website) {
+            foreach ($customers as $customer) {
+                try {
+                    $m2Customer = $this->customerRepository->get($customer['email'], $website->getId());
+
+                    $output->write(
+                        "<info>Customer {$customer['braintree_id']} found in {$website->getName()}...</info>"
+                    );
+
+                    foreach ($customer['storedCards'] as $storedCard) {
```

Comment:
should this be wrapped in a SQL commit?

Reply:
As discussed, the repository `save` method utilises transactions, and re-running the command will not result in duplicate data.